docs/proposals/federated-ruler.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Ruler Tenant Federation"
+linkTitle: "Ruler Tenant Federation"
+weight: 1
+slug: "Ruler Tenant Federation
+---
+
+- Author: [Rees Dooley](https://github.com/rdooley)
+- Date: September 2021
+- Status: Accepted
+
+## Overview
+
+This document aims to describe how to implement the ability to allow rules to cover data from more than a single Cortex tenant, here after refered to as federated rules. Since currently rules are owned by, query data from and save resulting series in the same tenant, this document aims to provide clear delineation of who owns a federated rule, what tenants the federated rule queries data from and where the federated rule saves resulting series.
+
+## Reasoning
+
+There are two primary use cases for allowing federated rules which query data from multiple tenants; administration of cortex and composite tenants.
+
+In the case of the administration of cortex, when running Cortex within a large organization, there may be metrics spanning across tenants which might be desired to be monitored e.g. administrative metrics of the cortex system like `prometheus_rule_evaluation_failures_total` aggregated by `__tenant_id__`. In this case, a team e.g. `infra` may wish to be able to create a rule, owned by `infra` which queries multiple tenants `t0|t1|...|ti` and stores resulting series in `infra`. More generally a tenant `A` may wish to query data from other tenants `B|C|D` and store the data in a tenant which may be `A`, but could be `E`.
+
+Additionally an organization may wish to treat several cortex tenants `t0|t1|...|ti` as one logical tenant due to compactor scalability. In this case a composite tenant `t0|t1|...|ti` would own a federated rule, which queries each subtenant `t0` thru `ti`. The resulting data could be sent to a specific tenant e.g. `admin` or as the composite tenant treats each subtenant uniformly, a random, but consistent subtenant can be chosen to store the resulting series in, in this case `tj` where 0 <= j <= i.
```

Comment:
clarification: the resulting series are spread approximately equally among the tenants, or all go to a single tenant?

Reply:
A recording rule's produced series should always go to the same single tenant, however if another recording rule produces series, those may be sent to a different single tenant. 

I'll add some two examples here to make this clearer